discovery-provider/tests/test_challenges.py

```diff
@@ -0,0 +1,369 @@
+        # to ensure that nothing happens
+        my_challenge.process(session, "test_event", [
+            {
+                "user_id": 1,
+                "block_number": 99
+            },
+        ])
+        session.flush()
+        actual = session.query(UserChallenge).filter(
+            UserChallenge.challenge_id == 'test_challenge_1',
+            UserChallenge.user_id == 1
+        ).first()
+        expected = {
+            "challenge_id": "test_challenge_1",
+            "user_id": 1,
+            "specifier": '1',
+            "is_complete": False,
+            "current_step_count": 1
+        }
+        assert model_to_dictionary(actual) == expected
+
+        # Now process events and make sure things change as expected
+        my_challenge.process(session, "test_event", [
+            {
+                "user_id": 1,
+                "block_number": 100
+            },
+            {
+                "user_id": 2,
+                "block_number": 100
+            },
+            {
+                "user_id": 3,
+                "block_number": 100
+            },
+            {
+                "user_id": 6,
+                "block_number": 100
+            }
+        ])
+        session.flush()
+
+        updated_complete = session.query(UserChallenge).filter(
+            UserChallenge.challenge_id == 'test_challenge_1'
+        ).all()
+        res_dicts = list(map(model_to_dictionary, updated_complete))
+        expected = [
+            # Should have incremented step count + 1
+            {
+                "challenge_id": "test_challenge_1",
+                "user_id": 1,
+                "specifier": '1',
+                "is_complete": False,
+                "current_step_count": 2
+            },
+            # Should be unchanged b/c it was already complete
+            {
+                "challenge_id": "test_challenge_1",
+                "user_id": 2,
+                "specifier": '2',
+                "is_complete": True,
+                "current_step_count": 3
+            },
+            # Should be newly complete
+            {
+                "challenge_id": "test_challenge_1",
+                "user_id": 3,
+                "specifier": '3',
+                "is_complete": True,
+                "current_step_count": 3
+            },
+            # Should be untouched bc user 5 wasn't included
+            {
+                "challenge_id": "test_challenge_1",
+                "user_id": 5,
+                "specifier": '5',
+                "is_complete": False,
+                "current_step_count": 2
+            },
+            # Should have created a brand new user 6
+            {
+                "challenge_id": "test_challenge_1",
+                "user_id": 6,
+                "specifier": '6',
+                "is_complete": False,
+                "current_step_count": 1
+            }
+        ]
+        assert expected == res_dicts
+
+def test_profile_completion_challenge(app):
```

Comment:
feels like this should be a sep file. maybe /tests/challenges/test_profile_completion_challenge.py?